### PR TITLE
[Feature] Support Objects365 Dataset

### PR DESCRIPTION
**Pull request by BIGWangYuDong**
Support Objects365v1 and Objects365v2

### CODE REVIEW

File: configs/objects365/faster_rcnn_r50_fpn_1x_obj365v1.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+_base_ = [
+    '../_base_/models/faster_rcnn_r50_fpn.py',
+    '../_base_/datasets/objects365v1_detection.py',
+    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
+]
+
+model = dict(roi_head=dict(bbox_head=dict(num_classes=365)))
+
+# Using 32 GPUS while training
```
Comment (ZwwWayne) [inline configs/objects365/faster_rcnn_r50_fpn_1x_obj365v1.py:9]: then we should modify the config name, and update the auto_scale_lr config

File: configs/objects365/faster_rcnn_r50_fpn_syncbn_1350k_obj365v1.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+model = dict(
+    backbone=dict(norm_cfg=norm_cfg),
+    roi_head=dict(bbox_head=dict(num_classes=365)))
+
+# Using 8 GPUS while training
+optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
+optimizer_config = dict(
+    _delete_=True, grad_clip=dict(max_norm=35, norm_type=2))
+
+runner = dict(type='IterBasedRunner', max_iters=1350000)
```
Comment (ZwwWayne) [inline configs/objects365/faster_rcnn_r50_fpn_syncbn_1350k_obj365v1.py:16]: Add comments to indicate the real epoch number

File: mmdet/datasets/objects365.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+            ann_file (str): Path of annotation file.
+
+        Returns:
+            list[dict]: Annotation info from COCO api.
+        """
+
+        self.coco = COCO(ann_file)
+        # The order of returned `cat_ids` will not
+        # change with the order of the CLASSES
+        self.cat_ids = self.coco.get_cat_ids(cat_names=self.CLASSES)
```
Comment (ZwwWayne) [inline mmdet/datasets/objects365.py:211]: it could be better to prepare unit tests

File: mmdet/datasets/objects365.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+        Args:
+            ann_file (str): Path of annotation file.
+
+        Returns:
+            list[dict]: Annotation info from COCO api.
+        """
+
+        self.coco = COCO(ann_file)
+        # 'categories' list in objects365_train.json an objects365_val.
```
Comment (BIGWangYuDong) [inline mmdet/datasets/objects365.py:99]: and

File: mmdet/datasets/objects365.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os.path as osp
+
+from .api_wrappers import COCO
+from .builder import DATASETS
+from .coco import CocoDataset
+
+# images exist in annotations but not in image folder.
+objv2_ignore_list = [
```
Comment (RangiLyu) [inline mmdet/datasets/objects365.py:9]: Is this a general problem? Or only exist in our data?

Comment (BIGWangYuDong) [reply mmdet/datasets/objects365.py:9]: It's a general problem

File: configs/objects365/faster_rcnn_r50_fpn_16x4_1x_obj365v1.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+lr_config = dict(
+    policy='step',
+    warmup='linear',
+    warmup_iters=1000,
+    warmup_ratio=1.0 / 1000,
+    step=[8, 11])
+
+# NOTE: `auto_scale_lr` is for automatically scaling LR,
+# USER SHOULD NOT CHANGE ITS VALUES.
+# base_batch_size = (32 GPUs) x (2 samples per GPU)
```
Comment (BIGWangYuDong) [inline configs/objects365/faster_rcnn_r50_fpn_16x4_1x_obj365v1.py:24]: 16 x 4

File: configs/objects365/faster_rcnn_r50_fpn_16x4_1x_obj365v2.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+lr_config = dict(
+    policy='step',
+    warmup='linear',
+    warmup_iters=1000,
+    warmup_ratio=1.0 / 1000,
+    step=[8, 11])
+
+# NOTE: `auto_scale_lr` is for automatically scaling LR,
+# USER SHOULD NOT CHANGE ITS VALUES.
+# base_batch_size = (32 GPUs) x (2 samples per GPU)
```
Comment (RangiLyu) [inline configs/objects365/faster_rcnn_r50_fpn_16x4_1x_obj365v2.py:24]: 16x4

**Review by RangiLyu — COMMENTED** [review]
LGTM. Can be merged after fixing typo.

File: configs/objects365/README.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  author={Shao, Shuai and Li, Zeming and Zhang, Tianyuan and Peng, Chao and Yu, Gang and Zhang, Xiangyu and Li, Jing and Sun, Jian},
+  booktitle={Proceedings of the IEEE/CVF international conference on computer vision},
+  pages={8430--8439},
+  year={2019}
+}
+```
+
+## Prepare Dataset
+
+1. You need to download and extract Objects365 dataset.
```
Comment (ZwwWayne) [inline configs/objects365/README.md:44]: Should indicate from which link or provide download script.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**